lib/acme/client/chain_identifier.rb

```diff
@@ -0,0 +1,37 @@
+class Acme::Client
+  class ChainIdentifier
+    def initialize(pem_certificate_chain)
+      @pem_certificate_chain = pem_certificate_chain
+    end
+
+    def match_name?(name)
+      issuers.any? do |issuer|
+        issuer.include?(name)
+      end
+    end
+
+    private
+
+    def issuers
+      x509_certificates.map(&:issuer).map(&:to_s)
+    end
+
+    def x509_certificates
+      @x509_certificates ||= splitted_pem_certificates.map { |pem| OpenSSL::X509::Certificate.new(pem) }
+    end
+
+    def splitted_pem_certificates
+      certificates = []
+      certificate_lines = []
+      @pem_certificate_chain.each_line do |line|
+        certificate_lines << line
+        if line.include?('END CERTIFICATE')
+          certificates << certificate_lines.join
+          certificate_lines = []
+        end
+      end
+
+      certificates
```

Comment:
```suggestion
    def splitted_pem_certificates
      @pem_certificate_chain.each_line.slice_after(/END CERTIFICATE/)
```

Reply:
I like that

Reply:
Okay, I didn't expect Ruby 2.1 still be on the support list, given its official security maintenance phase is long over.